apps/condo/domains/notification/tasks/deliverMessage.js

```diff
@@ -46,130 +51,225 @@ async function _sendMessageByAdapter (transport, adapter, messageContext) {
     return await adapter.send(messageContext)
 }
 
-// TODO(pahaz): we should chose the best transport for the message.
-//  We can chose transport depends on the message.type?
-//  or use something like message.user.profile.preferredNotificationTransport if user want to get messages from TG
+/**
+ * Requests User schema for a record and returns contacts if found
+ * @param context
+ * @param userId
+ * @returns {Promise<{}>}
+ * @private
+ */
+async function _getUserContacts (context, userId) {
+    const contacts = {}
+    const userData = await UserAdmin.getOne(context, { id: userId })
+    const userEmail = get(userData, 'email')
+    const userPhone = get(userData, 'phone')
+
+    if (userEmail) contacts.email = userEmail
+    if (userPhone) contacts.phone = userPhone
+
+    return contacts
+}
 
 /**
- * Calculates transport types priority queue for a message according to provided message data,
- * and fallback transports if more prioritized transports fail message delivery.
+ * Combines message & user contacts matching priorities
  * @param message
- * @returns {Promise<string[]>}
+ * @param userContacts
+ * @returns {{}}
  * @private
  */
-async function _choseMessageTransport (message) {
-    const { phone, user, email, id } = message
-    const transports = []
+function _prepareContacts (message = {}, userContacts = {}) {
+    const userId = get(message, ['user', 'id'])
+    const contacts = { ...userContacts }
 
-    // if message has phone field, SMS would be the only priority transport
-    if (!isEmpty(phone)) return [SMS_TRANSPORT]
+    if (message.phone) contacts.phone = message.phone
+    if (message.email) contacts.email = message.email
+    if (message.remoteClient) contacts.remoteClient = message.remoteClient
+    if (userId) contacts.userId = userId
 
-    // if message doesn't have phone, but has email field, EMAIL would be the only priority transport
-    if (!isEmpty(email)) return [EMAIL_TRANSPORT]
+    return contacts
+}
 
-    // if user is provided, we can try to send PUSH notifications wither a priority transport
-    // if phone & email are absent, or fallback transport if phone & email are present but fail to deliver message
-    // at the moment we don't want to use fallback!
-    if (!isEmpty(user)) {
-        transports.push(PUSH_TRANSPORT)
-    }
+/**
+ * Detects which transports message has contacts for
+ * @param message
+ * @returns {Set<unknown>}
+ * @private
+ */
+function _getMessageTransports (message) {
```

Comment:
you should change this logic and get it from the https://github.com/open-condo-software/condo/pull/2171/files